crates/mcp/src/surfpool/start_surfnet.rs

```diff
@@ -0,0 +1,151 @@
+use serde::Serialize;
+use surfpool_core::{start_local_surfnet, surfnet::svm::SurfnetSvm};
+use surfpool_types::{SimnetConfig, SimnetEvent, SurfpoolConfig};
+
+#[derive(Serialize)]
+pub struct StartSurfnetResponse {
+    pub success: Option<StartSurfnetSuccess>,
+    pub error: Option<String>,
+}
+
+#[derive(Serialize)]
+pub struct StartSurfnetSuccess {
+    pub surfnet_url: String,
+    pub surfnet_id: u16,
+}
+
+impl StartSurfnetResponse {
+    pub fn success(data: StartSurfnetSuccess) -> Self {
+        Self {
+            success: Some(data),
+            error: None,
+        }
+    }
+
+    pub fn error(message: String) -> Self {
+        Self {
+            success: None,
+            error: Some(message),
+        }
+    }
+}
+
+pub fn run(surfnet_id: u16) -> StartSurfnetResponse {
+    let (surfnet_svm, simnet_events_rx, geyser_events_rx) = SurfnetSvm::new();
+
+    let (simnet_commands_tx, simnet_commands_rx) = crossbeam_channel::unbounded();
+    let (subgraph_commands_tx, _subgraph_commands_rx) = crossbeam_channel::unbounded();
+
+    let simnet_events_tx = surfnet_svm.simnet_events_tx.clone();
+
+    let mut config = SurfpoolConfig::default();
+    let rpc_port = 8899 + 10000 * surfnet_id;
+    let ws_port = 8890 + 10000 * surfnet_id;
```

Comment:
Using only the `surfnet_id` to choose the port is insufficient. If the user has started a surfnet _without_ the MCP, then starts another with the MCP, they will have a port collision.

Add a helper function to ping `*8899` ports to find the next one that's open. Call this function _before_ entering this `run` function. Then in our `pub surfnets: Arc<RwLock<Vec<u16>>>` field, we should instead store a mapping of id to port.

So whenever a surfnet is going to be started we:
 1. Get the next ID
 2. Get the next open port
 3. Call `start_surfnet::run`, passing in the ports it should run on
 4. Store the id<>port mapping in the `surfnets` lock